From dbd1a59b239b3902e717fdeb063883dbb0b06ee9 Mon Sep 17 00:00:00 2001
From: Adrian Perez de Castro <aperez@igalia.com>
Date: Sun, 26 May 2024 14:24:35 -0700
Subject: [PATCH 1/2] Remove ARM-specific declarations in FELighting.h unneeded
 after 272873@main

Unreviewed build fix.

* Source/WebCore/platform/graphics/filters/FELighting.h: Remove unneeded
  declarations for the getPowerCoefficients() and platformApplyNeon()
  functions, which are now defined elsewhere; and were causing a build
  failure due to usage of the protected LightingData type.

Canonical link: https://commits.webkit.org/279334@main

Backport this patch for fixing following compile error:
webkitgtk-2.44.1/Source/WebCore/platform/graphics/filters/FELighting.h:73:41: error: 'LightingData' does not name a type 
   73 |     inline void platformApplyNeon(const LightingData&, const LightSource::PaintingData&);

Upstream-Status: Backport [https://github.com/WebKit/WebKit/commit/36d1b5d7c0ef9a733ee8055b1f35b1d24435d538]
Signed-off-by: Changqing Li <changqing.li@windriver.com>
---
 Source/WebCore/platform/graphics/filters/FELighting.h | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/Source/WebCore/platform/graphics/filters/FELighting.h b/Source/WebCore/platform/graphics/filters/FELighting.h
index 4efab920..dcd80b6f 100644
--- a/Source/WebCore/platform/graphics/filters/FELighting.h
+++ b/Source/WebCore/platform/graphics/filters/FELighting.h
@@ -68,11 +68,6 @@ protected:
 
     std::unique_ptr<FilterEffectApplier> createSoftwareApplier() const override;
 
-#if CPU(ARM_NEON) && CPU(ARM_TRADITIONAL) && COMPILER(GCC_COMPATIBLE)
-    static int getPowerCoefficients(float exponent);
-    inline void platformApplyNeon(const LightingData&, const LightSource::PaintingData&);
-#endif
-
     Color m_lightingColor;
     float m_surfaceScale;
     float m_diffuseConstant;
-- 
2.25.1

